internal/health/health.go

```diff
@@ -0,0 +1,186 @@
+package health
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kubeshop/botkube/internal/httpx"
+	"github.com/kubeshop/botkube/internal/plugin"
+	"github.com/kubeshop/botkube/pkg/bot"
+	"github.com/kubeshop/botkube/pkg/config"
+	"github.com/kubeshop/botkube/pkg/notifier"
+)
+
+type BotkubeStatus string
+
+const (
+	BotkubeStatusHealthy   BotkubeStatus = "Healthy"
+	BotkubeStatusUnhealthy BotkubeStatus = "Unhealthy"
+)
+
+const (
+	healthEndpointName = "/healthz"
+)
+
+// Checker gives health bot agent status.
+type Checker struct {
+	applicationStarted bool
+	ctx                context.Context
+	config             *config.Config
+	pluginHealthStats  *plugin.HealthStats
+	bots               map[string]bot.Bot
+	sinks              []notifier.Sink
+}
+
+type pluginStatuses struct {
+	Enabled  bool
+	Status   string
+	Restarts string
+}
+
+type botStatus struct {
+	Status BotkubeStatus
+}
+
+type platformStatuses map[string]bot.Status
+
+// Status defines bot agent status.
+type Status struct {
+	Botkube   botStatus
+	Plugins   map[string]pluginStatuses
+	Platforms platformStatuses
+}
```

Comment:
you should specify the `json` tags, otherwise it uses the fields names, which looks odd:
`{"Botkube":{"Status":"Unhealthy"},"Plugins":{},"Platforms":{}}`

You should:
1. Move the `bot.Status` object into this package. Use it bot for sink and bots. Add JSON tags.
2. Add JSON tags to `Status` with `omittempty`:
    ```go
    // Status defines bot agent status.
    type Status struct {
	    Botkube   botStatus                 `json:"botkube"`
	    Plugins   map[string]pluginStatuses `json:"plugins,omitempty"`
	    Platforms platformStatuses          `json:"platforms,omitempty"`
    }
    
    ```